File: Python/2810.py
# 2810. Faulty Keyboard
# https://leetcode.com/problems/faulty-keyboard/

class Solution:
  def finalString(self, s: str) -> str:
    out = ""

    for c in s:
      if c != "i":
        out += c
      else:
        out = out[::-1]

    return out

File: Rust/2810.rs
/**
 * 2810. Faulty Keyboard
 * https://leetcode.com/problems/faulty-keyboard/
 */

impl Solution {
  pub fn final_string(s: String) -> String {
    let mut out: String = String::new();

    for c in s.chars() {
      if c != 'i' {
        out.push(c);
      } else {
        out = out.chars().rev().collect::<String>();
      }
    }

    return out;
  }
}
